manifests/service.pp

```diff
@@ -28,53 +28,49 @@
   $catalina_home  = undef,
   $catalina_base  = undef,
   $use_jsvc       = false,
+  $use_init       = false,
   $java_home      = undef,
   $service_ensure = running,
   $service_enable = undef,
-  $use_init       = false,
   $service_name   = undef,
   $start_command  = undef,
   $stop_command   = undef,
-  $user           = $::tomcat::user,
+  $user           = undef,
 ) {
+  include tomcat
+  $_user = pick($user, $::tomcat::user)
```

Comment:
The `tomcat::service` define cannot inherit the tomcat class, so `$::tomcat::user` may not be available at parameter default time so it must be brought into the definition; you can see the `include tomcat` line right above this.